System/Random.hs

```diff
@@ -1083,6 +1085,22 @@ uniformIntegerM (l, h) gen
         let v' = v * b + fromIntegral x
         v' `seq` f (mag * b) v'
 
+unbiasedIntMult32 :: MonadRandom g m => Word32 -> g -> m Word32
```

Comment:
@idontgetoutmuch Can you add a link to a post or a reference to a paper were this algorithm came from?

Reply:
One more thing, function name is a bit misleading it works on `Word` but contains an `Int`, shouldn't it be `unbiasedWordMult32`?

Reply:
All done and I replaced  `>>= return . ...` with `<$>` to fit with the house style :)